testing/time_test.ts

```diff
@@ -632,3 +634,88 @@ Deno.test("Date from FakeTime is structured cloneable", () => {
   assert(date instanceof Date);
   assert(cloned instanceof Date_);
 });
+
+Deno.test("Faked timer functions throws when called after FakeTime is restored", () => {
+  let fakeSetTimeout: typeof setTimeout;
+  let fakeClearTimeout: typeof clearTimeout;
+  let fakeSetInterval: typeof setInterval;
+  let fakeClearInterval: typeof clearInterval;
+  {
+    using _time: FakeTime = new FakeTime();
+    fakeSetTimeout = setTimeout;
+    fakeClearTimeout = clearTimeout;
+    fakeSetInterval = setInterval;
+    fakeClearInterval = clearInterval;
+  }
+  assertThrows(() => fakeSetTimeout(() => {}, 0), TimeError, "no fake time");
+  assertThrows(() => fakeClearTimeout(0), TimeError, "no fake time");
+  assertThrows(() => fakeSetInterval(() => {}, 0), TimeError, "no fake time");
+  assertThrows(() => fakeClearInterval(0), TimeError, "no fake time");
+});
+
+Deno.test("Faked Date.now returns real time after FakeTime is restored", () => {
+  let fakeDateNow: typeof Date.now;
+  {
+    using _time: FakeTime = new FakeTime();
+    fakeDateNow = Date.now;
+  }
+  assertAlmostEquals(Date.now(), fakeDateNow());
+});
+
+Deno.test("FakeTime can be called multiple times (later call is effective)", () => {
+  const now = Date.now();
+  try {
+    const _time1 = new FakeTime(1000);
+    const _time2 = new FakeTime(2000);
+    assertEquals(Date.now(), 2000);
+  } finally {
+    FakeTime.restore();
+  }
```

Comment:
This should work once #5125 is merged.
```suggestion
  {
    using _time1 = new FakeTime(1000);
    using _time2 = new FakeTime(2000);
    assertEquals(Date.now(), 2000);
  }
```

Reply:
This case has been removed